era-compiler-solidity/src/yul/parser/statement/for_loop.rs

```diff
@@ -10,27 +10,36 @@ use crate::yul::error::Error;
 use crate::yul::lexer::token::location::Location;
 use crate::yul::lexer::token::Token;
 use crate::yul::lexer::Lexer;
+use crate::yul::parser::dialect::era::EraDialect;
+use crate::yul::parser::dialect::Dialect;
 use crate::yul::parser::statement::block::Block;
 use crate::yul::parser::statement::expression::Expression;
 
 ///
 /// The Yul for-loop statement.
 ///
-#[derive(Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
-pub struct ForLoop {
+#[derive(Debug, serde::Serialize, serde::Deserialize, Clone, PartialEq, Eq)]
+#[serde(bound = "P: serde::de::DeserializeOwned")]
```

Comment:
No, unfortunately, I have tried to do it without this macro but the compiler rejects it with an error: 

```
error[E0283]: type annotations needed: cannot satisfy `P: Deserialize<'_>`
  --> era-compiler-solidity/src/yul/parser/statement/for_loop.rs:22:12
   |
22 | pub struct ForLoop<P>
   |            ^^^^^^^^^^
   |
note: multiple `impl`s or `where` clauses satisfying `P: Deserialize<'_>` found
  --> era-compiler-solidity/src/yul/parser/statement/for_loop.rs:21:35
   |
21 | #[derive(Debug, serde::Serialize, serde::Deserialize, Clone, PartialEq, Eq)]
   |                                   ^^^^^^^^^^^^^^^^^^
...
24 |     P: Dialect + serde::de::DeserializeOwned,
   |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: required for `ForLoop<P>` to implement `Deserialize<'de>`
  --> era-compiler-solidity/src/yul/parser/statement/for_loop.rs:21:35
   |
21 | #[derive(Debug, serde::Serialize, serde::Deserialize, Clone, PartialEq, Eq)]
   |                                   ^^^^^^^^^^^^^^^^^^ unsatisfied trait bound introduced in this `derive` macro
22 | pub struct ForLoop<P>
   |            ^^^^^^^^^^
   = note: this error originates in the derive macro `serde::Deserialize` (in Nightly builds, run with -Z macro-backtrace for more info)


```

Maybe this could be worked around by implementing it manually instead of asking to `derive` it.